### PR TITLE
Add doc: Update from HamburgerMenu to NavigationView

### PR DESCRIPTION
**Pull request by mvegaca**
Need documentation that explains how to update a previously generated project #2082

### CODE REVIEW

File: docs/projectTypes/updatetonavigationview.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    public ICommand ItemInvokedCommand => _itemInvokedCommand ?? (_itemInvokedCommand = new RelayCommand<NavigationViewItemInvokedEventArgs>(OnItemInvoked));
+
+    public ShellViewModel()
+    {
+    }
+
+    public void Initialize(Frame frame, NavigationView navigationView)
+    {
+        _navigationView = navigationView;
+        NavigationService.Frame = frame;
```
Comment (wind-rider) [inline docs/projectTypes/updatetonavigationview.md]: Where does the NavigationService property originate from? Previously this was injected using dependency injection with the Prism framework

File: docs/projectTypes/updatetonavigationview.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    private object _selected;
+    private ICommand _itemInvokedCommand;
+
+    public object Selected
+    {
+        get { return _selected; }
+        set { Set(ref _selected, value); }
+    }
+
+    public ICommand ItemInvokedCommand => _itemInvokedCommand ?? (_itemInvokedCommand = new RelayCommand<NavigationViewItemInvokedEventArgs>(OnItemInvoked));
```
Comment (wind-rider) [inline docs/projectTypes/updatetonavigationview.md]: RelayCommand does not exist in Prism

File: docs/projectTypes/updatetonavigationview.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+ShellViewModel will need the NavigationView instance (explained below), you will have to add it on initialization.
+
+### C# code you will have to modify:
+ - Add the navigationView control to ViewModel initialization.
+
+The resulting code should look like this:
+ ```csharp
+public sealed partial class ShellPage : Page
+{
+    public ShellViewModel ViewModel { get; } = new ShellViewModel();
```
Comment (wind-rider) [inline docs/projectTypes/updatetonavigationview.md]: `new ShellViewModel()` does not work with dependency injection on Prism

File: docs/projectTypes/updatetonavigationview.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
+# Update from HamburgerMenu to NavigationView
+If you have an UWP project created with WTS with the project type **NavigationPane** and you want to update to NavigationView, please follow these steps:
+
+## 1. Update min target version in project properties
+NavigationView is a Fall Creators Update control, to start using it in your project is neccessary that  you set FCU as min version.
+![](../resources/project-types/fcu-min-version.png)
+
+## 2. Update ShellPage.xaml
+The updated ShellPage will include the NavigationView and add the MenuItems directly in Xaml. The NavigationViewItems include an extension property that contains the target page type to navigate in the frame.
```
Comment (wind-rider) [inline docs/projectTypes/updatetonavigationview.md]: Why not dynamically anymore through `MenuItemsSource`?

**Comment by mrlacey** [comment]
@wind-rider this document only currently covers the MVVMBasic framework.
That it doesn't call this out or cover other frameworks is one of the things that need addressing before it goes live.

File: docs/projectTypes/updatetonavigationview.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+- [Framework: MVVM Light, Language: C#](updatetonavigationview/mvvmlight-cs.md)
+- [Framework: MVVMBasic, Language: C#](updatetonavigationview/mvvmbasic-cs.md)
+- [Framework: Caliburn.Micro, Language: C#](updatetonavigationview/caliburnmicro-cs.md)
+- [Framework: Prism, Language: C#](updatetonavigationview/prism-cs.md)
+
+- [Framework: Codebehind, Language: VB](updatetonavigationview/codebehind-vb.md)
+- [Framework: MVVM Light, Language: VB](updatetonavigationview/mvvmlight-vb.md)
+- [Framework: MVVMBasic, Language: VB](updatetonavigationview/mvvmbasic-vb.md)
+
+If you use actions on your menu, also look at this updated [document](navigationpane.md#invokecode)
```
Comment (mrlacey) [inline docs/projectTypes/updatetonavigationview.md]: This doc also needs updating.

**Comment by mvegaca** [comment]
@dgomezc  and I just finish docs edition. Can you add a review? @mrlacey @sibille

**Comment by OzBob** [comment]
This Hamburger migration to NavigationView doesn't seem to be in the WTS templates nightly build.  

https://github.com/microsoft/WindowsTemplateStudio/blob/dev/templates/Wpf/_comp/_shared/Project.SplitView/Views/ShellWindow.xaml#L86

**Pull request merged**